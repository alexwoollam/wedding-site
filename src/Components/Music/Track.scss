@import '../../variables';

.track-card{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  min-height: 540px;
  &-button{
    background-color: $dustygreen;
    &:hover{
      background-color: $dustygreen--dark;
    }
  }
  &-body{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    padding: 0px;
    min-width: 100%;

    & .track-card-preview{
      align-self: end;
      min-width: 100%;
    }
  }
}

.cover-preview{
  width: 280px;
  height: 280px;
  border-radius: 5px;

}