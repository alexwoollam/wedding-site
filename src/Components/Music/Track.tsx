import React, {useState} from 'react';
import ReactAudioPlayer from 'react-audio-player';
import {Container, Card, Row, Col, CardImg, CardBody, CardTitle, CardSubtitle, Button} from 'reactstrap';

interface PropsDefined {
    data: object,
    playTrack?: any,
    addTrack?: any,
    isSelected?: boolean,
}

function Track( props: PropsDefined ){

    const [playingTrack, setPlayingTrack] = useState<string>('null');

    if( ! props ){ 
        return null;
    }
    let items = Object.values(props.data);


    if( items[1] ){        
        return (
            <Container className="container mx-auto">
                <Row>
                    {items[1].map( (track, index)=>(
                        <Col sm={12} md={4}>
                            <Card className="box-border m-3 p-2 border-4 rounded">
                                <CardImg src={ track.album.images[1].url } alt={"artwork for " + track.name } />
                                <Button onClick={ () => props.addTrack( track.id, track.name, track.artists[0].name, track.album.images[2].url ) } >Add track</Button>
                                <CardBody>
                                    
                                            <CardTitle tag="h4">{ track.name }</CardTitle>
                                            <CardSubtitle tag="h5">{ track.artists[0].name }</CardSubtitle>
                                            <br/>
                                            {
                                            track.preview_url ? 
                                            (
                                                <ReactAudioPlayer
                                                className={'w-100'}
                                                src={track.preview_url}
                                                controls
                                                />
                                            ):
                                            null
                                            }
                                    
                                </CardBody>
                            </Card>
                        </Col>
                    ))}
                </Row>
            </Container>
        );
    } else {
        return null;
    }
    
} 

export default Track;