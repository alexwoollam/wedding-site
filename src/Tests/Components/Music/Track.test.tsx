//ts-noscan
import '@testing-library/jest-dom';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Track from '../../../Components/Music/Track';


Enzyme.configure({ adapter: new Adapter() });

describe('<Track/> component', () => {

    let MockTracks = new Object();
    MockTracks = [
            {'some string':'string'},
            [
                {
                    'album': {
                        'images': [
                            {'url': "https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"},
                            {'url': "https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"},
                            {'url': "https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"},
                        ]
                    },
                    'artists': [{
                        'name': 'Blur' 
                    }],
                    'id': "1FTSo4v6BOZH9QxKc3MbVM",
                    'name': "Song 2",
                    'preview_url': "https://p.scdn.co/mp3-preview/183c0855e94b58dcb267e2b0721d4a3c99260acf?cid=1686551f71a7417da335f1cb88526648",
                },
                {
                    'album': {
                        'images': [
                            {'url': "https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"},
                            {'url': "https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"},
                            {'url': "https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"},
                        ]
                    },
                    'artists': [{
                        'name': 'Nirvana' 
                    }],
                    'id': "1FTSo4v6BOZH9QxKc3MbVM",
                    'name': "Bleach",
                    'preview_url': "https://p.scdn.co/mp3-preview/183c0855e94b58dcb267e2b0721d4a3c99260acf?cid=1686551f71a7417da335f1cb88526648",
                },
            ]
        ];


    test('Component should reuturn null if no props passed', () => {
        const comp = shallow(<Track />);
        expect(comp.type()).toEqual(null)
    });

    test('Track title should be Song 2', () => {
        render(<Track data={ MockTracks } />);
        const linkElement = screen.getByText('Song 2');
        expect(linkElement).toBeInTheDocument();
    });

    test('Artist should be blur', () => {
        render(<Track data={ MockTracks }/>);
        const linkElement = screen.getByText('Blur');
        expect(linkElement).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const tree = renderer
        .create(<Track data={ MockTracks } />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
