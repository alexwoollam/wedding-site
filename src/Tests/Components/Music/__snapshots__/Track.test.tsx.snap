// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track/> component renders correctly 1`] = `
<div
  className="container mx-auto container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-4"
    >
      <div
        className="box-border m-3 p-2 border-4 rounded card"
      >
        <img
          alt="artwork for Song 2"
          className="card-img"
          src="https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"
        />
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Add track
        </button>
        <div
          className="card-body"
        >
          <h4
            className="card-title"
          >
            Song 2
          </h4>
          <h5
            className="card-subtitle"
          >
            Blur
          </h5>
          <br />
          <div
            data-testid="preview-button"
          >
            <audio
              autoPlay={false}
              className="react-audio-player w-100"
              controls={true}
              id=""
              loop={false}
              muted={false}
              preload="metadata"
              src="https://p.scdn.co/mp3-preview/183c0855e94b58dcb267e2b0721d4a3c99260acf?cid=1686551f71a7417da335f1cb88526648"
              style={Object {}}
              title="https://p.scdn.co/mp3-preview/183c0855e94b58dcb267e2b0721d4a3c99260acf?cid=1686551f71a7417da335f1cb88526648"
            >
              <p>
                Your browser does not support the 
                <code>
                  audio
                </code>
                 element.
              </p>
            </audio>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-md-4"
    >
      <div
        className="box-border m-3 p-2 border-4 rounded card"
      >
        <img
          alt="artwork for Bleach"
          className="card-img"
          src="https://i.scdn.co/image/ab67616d00001e02d99d0823b831a78b50379100"
        />
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Add track
        </button>
        <div
          className="card-body"
        >
          <h4
            className="card-title"
          >
            Bleach
          </h4>
          <h5
            className="card-subtitle"
          >
            Nirvana
          </h5>
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
